### ISSUE TITLE
Call flash.Lib.current.loaderInfo.parameters from haxe.web.Request.getParams()

### ISSUE THREAD

**Issue by ashes999** [HaxeFoundation/haxe#5158]
`haxe.web.Request.getParams()` returns a map of `<String, String>` with key/value pairs for each URL parameter. It only works on JS, Neko, and PHP.

`flash.Lib.current.loaderInfo.parameters` also returns a map of `<String, String>` with key/value pairs for each URL parameter.

Is there any reason not to add another clause to `haxe.web.Request`, similar to?:

```
#elseif flash
return flash.Lib.current.loaderInfo.parameters;
#end
```

... or something similar (if I've gotten the types wrong, and they return different types)?

**Comment by Simn** [comment]
Implement it, test it, send a PR and we'll see.

**Comment by ashes999** [comment]
Are there any instructions on how to run the unit/automated tests? I don't see anything in `Contributing.MD`.

**Comment by Simn** [comment]
```
cd tests/unit
haxe compile.hxml
```

However, haxe.web has no tests.

**Comment by ashes999** [comment]
I'll try it. I don't see much point though if there are no tests.  I don't think I'm comfortable back-filling tests for this.

I'll run the tests and open a PR and we can take it from there.

**Comment by Simn** [comment]
I was assuming that you have a use-case for this which allows you to test it.

**Comment by ashes999** [comment]
I do, but it's a strange one:

- Add this code somewhere and print out what it returns
- Build it on Flash
- Open the SWF in Chrome with query parameters (eg. ?a=1%b=hi%20there!&c=false)
- Inspect that the hash has the right key/value pairs

If you can help me write a test for this for Flash, I can definitely add it to my PR.

**Comment by Simn** [comment]
Sounds rather complicated indeed. There's no need to add it to our tests, as long as you tell me it works for you I'm fine with making the change because it's currently an error-case anyway (right?).

**Comment by ashes999** [comment]
Yes, that's right; without this, we get a compile-timer error on Flash.

**Comment by ashes999** [comment]
When I run `haxe compile.hxml` I get lots of errors similar to:

```
src/unit/TestType.hx:345: characters 20-22 : Unexpected is
src/unit/TestType.hx:345: characters 20-22 : Missing ;
src/unit/TestType.hx:345: characters 23-29 : Missing ;
src/unit/TestType.hx:345: characters 29-30 : Unexpected )
```

If I comment out the offending lines (eg. `t((c3.covariant() is Child2_1));`), I get another error:

```
src/unit/TestType.hx:612: lines 612-616 : Module haxe.Constraints does not define type Constructible
```

Am I supposed to use a specific branch? This is on `master`.

**Comment by Simn** [comment]
You can either use the 3.2.1 compiler with master or a nightly with development.

**Comment by Justinfront** [comment]
I know a bit about this topic or atleast flash.Lib.current.loaderInfo.parameters from working with as3 and some java developers who wanted to embed my movie in a bad way. I had to work round the problem several times for various clients.  

The problem is that flash will not populate these parameters imediately, the problem is to do with the initialization of the Main class and properies like width not being avaliable to the flash movie imediatately as they are provided by the browser.  With as2 the population is normally imediate as it was low tech but efficient. With as3 they introduced an object that flash player populates when it feels like it after it has run the main classes constructor, ie with lots of parameters this can be really slow! - It can take quite awhile depending on the exact form of flash embed used ( differing in the way you use SWFObject or similar with embedding in body and head acting differently ) and also vary significantly on the version of browser, I seem to remember specific versions of Internet Explorer being bad and some versions of firefox very good, chrome was normally in between, and new browsers were not always better.

I think really this is not an issue that should not be baked into Haxe unless it's possible to delay boot suitably when such code is used but given that it can be quite slow that seems like a bad idea, also there are similar issues when loading movies into others ( and dealing with application domains ) etc... it's really likely a minefield and one of the few times flash has the same downside as html.

My approach I documented here, one I used on many projects, Shelly did not like it, but practically it worked very well in many many projects.

http://old.haxe.org/doc/flash/byexample/flash/flash9/loaderinfo

I suggest that we just update this tutorial to the 'code cookbook' perhaps? Rather than try to fix a browser embed issue, since they really are beyond the control of Haxe code.
